### PR TITLE
Add Beats upgrade docs for 8.0

### PR DESCRIPTION
**Pull request by dedemorton**
Adds Beats upgrade docs for 8.0.

Preview link: https://beats_30355.docs-preview.app.elstc.co/guide/en/beats/libbeat/master/upgrading.html 

TODO:
- [x] Test the upgrade.
- [x] Mention that the Beats version must be <= Elasticsearch version (Beats cannot send to older versions of Elasticsearch). Confirm that this is true!

### CODE REVIEW

**Comment by dedemorton** [comment]
I did some some simple tests to verify the flow of steps in the document (installed 7.17 then upgraded to 8.0). This PR is ready for a final review.

**Comment by kvch** [comment]
The upgrade steps were tested manually. There were no issues.

**Review by kvch — APPROVED** [review]

File: libbeat/docs/upgrading.asciidoc [hunk 1]
```diff
@@ -116,201 +106,132 @@ you do not specify this setting, logs are stored in the directory you extracted
 the archive to.
 --
 
+//REVIEWERS: What (if anything) should we say here ^^ about the modules.d dir?
+//Is there a situation where users might have configs defined in 7.17, but also
+//want to use some new modules from 8.0? If so, what do we recommend they do?
+
```
Comment (dedemorton) [inline libbeat/docs/upgrading.asciidoc:112]:
```suggestion
```

File: libbeat/docs/upgrading.asciidoc [hunk 2]
```diff
@@ -116,201 +106,132 @@ you do not specify this setting, logs are stored in the directory you extracted
...
-After removing all references to old fields, you should set
-`migration.6_to_7.enabled: false` so that field aliases will not be created
-during your next minor upgrade.
-
-The aliases will be removed in 8.0.
-
-TIP: Did you run the Beat or load the index template before reading this section?
-That's OK. See the clean-up steps described under <<missing-fields>>.
+//REVIEWERS: Do users need to load any other assets besides the index template
+//and dashboards?
```
Comment (dedemorton) [inline libbeat/docs/upgrading.asciidoc:130]:
```suggestion
```

File: libbeat/docs/upgrading.asciidoc [hunk 3]
```diff
@@ -116,201 +106,132 @@ you do not specify this setting, logs are stored in the directory you extracted
...
 
-:asset: the index template
-:option: template
+If you're not collecting time series data, you can continue to use {beats}
+to send data to aliases and indices. To do this, create a custom index template
+and load it manually. To learn more about creating index templates, refer to
+{ref}/index-templates.html[Index templates].
 
-include::upgrade-setup-commands.asciidoc[]
+//REVIEWERS: Do users need to explicitly disable template loading?
```
Comment (dedemorton) [inline libbeat/docs/upgrading.asciidoc:154]:
```suggestion
```

File: libbeat/docs/upgrading.asciidoc [hunk 4]
```diff
@@ -116,201 +106,132 @@ you do not specify this setting, logs are stored in the directory you extracted
...
-=== Troubleshoot {beats} upgrade issues
+[discrete]
+[[missing-timestamp-field]]
+==== Timestamp field is missing
 
-This section describes common problems you might encounter when upgrading to
-{beats} 7.x.
+//REVIEWERS: I think this section could be improved by more detail about
+//the error message. What does the user see in the logs to indicate
+//this problem?
```
Comment (dedemorton) [inline libbeat/docs/upgrading.asciidoc:222]:
```suggestion
```

File: libbeat/docs/upgrading.asciidoc [hunk 5]
```diff
@@ -116,201 +106,132 @@ you do not specify this setting, logs are stored in the directory you extracted
...
+will be unable to index the event. To fix the problem, modify your processor
+configuration to avoid removing the timestamp field.
 
+[discrete]
 [[missing-fields]]
 ==== Missing fields or too many fields in the index
 
+//REVIEWERS: IS this still necessary? I think the defaults might
+//take care of this, but I'll need to do some testing.
+
```
Comment (dedemorton) [inline libbeat/docs/upgrading.asciidoc:234]:
```suggestion
```

**Pull request merged**